src/osd/PrimaryLogPG.cc

```diff
@@ -250,16 +250,19 @@ class PrimaryLogPG::C_OSD_AppliedRecoveredObjectReplica : public Context {
 };
 
 // OpContext
-void PrimaryLogPG::OpContext::start_async_reads(PrimaryLogPG *pg)
+void PrimaryLogPG::OpContext::start_async_reads(PrimaryLogPG *pg, CephContext* cct)
 {
   inflightreads = 1;
   list<pair<boost::tuple<uint64_t, uint64_t, unsigned>,
 	    pair<bufferlist*, Context*> > > in;
   in.swap(pending_async_reads);
+  struct OnReadComplete* porc = new OnReadComplete(pg, this);
+  generic_dout(20) << __func__
```

Comment:
i'd suggest revert this change. it's handy for debugging this issue, but it won't be that helpful after done with it.